### PR TITLE
Do JavaScript file updating in JavaScript

### PR DESCRIPTION
**Pull request by hmarr**
Similar to #36, but for updating rather than parsing.

Like #36 this relies heavily on the Yarn internals. However, the code that touches Yarn is relatively well commented, and duplicates the minimum possible from Yarn, so it shouldn't be too brittle (look through previous commits in this branch if you want to see what a more brittle implementation looks like 😉).

### Caveats

- The helper uses async/await, which is only available in Node 7. **Make sure you're running Node 7+** wherever you're using Bump Core.
- While the JS tests stub out HTTP requests to yarn, the Ruby tests use the JS helper as a subprocess, so can't do any HTTP stubbing. This means the Ruby test suite still fails when you're offline. This shouldn't be a blocker to merging - we're not making the situation any worse; we're just not making any better, either.

### What about the update checker?

As it stands, the Ruby implementation of the JavaScript update checker is fine. As node uses nested node_modules folders, there's no clever version resolution to be done when you're updating - you can simply go to the latest version.

### CODE REVIEW

**Comment by greysteil** [comment]
Super clear, and all makes sense to me. 👍

**Review by greysteil — APPROVED** [review]

File: helpers/javascript/lib/updater.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+// exactly the same as adding, except the package already exists in the
+// manifests.
+//
+// Usually, calling Add.init() would execute a series of steps: resolve, fetch,
+// link, run lifecycle scripts, cleanup, then save new manifest (package.json).
+// We only care about the first and last steps: resolve, then save the new
+// manifest. Fotunately, overriding bailout() gives us an opportunity to skip
+// over the intermediate steps in a relatively painless fashion.
+class LightweightAdd extends Add {
+  async bailout(patterns) {
```
Comment (waltfy) [inline helpers/javascript/lib/updater.js]: My lack of familiarity with the internals of `yarn` makes me want to see more context about `bailout` here, specifically how it fits into the Add flow of execution? @hmarr thoughts?

File: helpers/javascript/lib/updater.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    {},
+    manifest.peerDependencies,
+    manifest.optionalDependencies,
+    manifest.devDependencies,
+    manifest.dependencies
+  );
+}
+
+async function updateDependencyFiles(directory, depName, desiredVersion) {
+  // Setup for Yarn internals
```
Comment (waltfy) [inline helpers/javascript/lib/updater.js]: I'd rather see a comment that addresses why we need this setup? Or if is not that important just remove it?

**Comment by hmarr** [comment]
@waltfy just rebased since #42 was merged, so this is slightly smaller now :-)

**Review by waltfy — APPROVED** [review]
Nice work! Two minor things I picked up on, other than that looks great!

File: helpers/javascript/lib/updater.js (hunk 1 above)
Comment (hmarr) [reply helpers/javascript/lib/updater.js]: @waltfy I added more comments - would you mind reading them (and the comments already there) and checking they make sense to you?

**Pull request merged**